### DISCUSSION TITLE
Deepmerge-ts causing error on build

### DISCUSSION THREAD

**Discussion by fatihaziz** [RebeccaStevens/deepmerge-ts#179] [Q&A] [answered]
So the error is this 
![image](https://user-images.githubusercontent.com/23287658/186348035-fb6bd379-1002-4a17-bac2-016e6a03bef1.png)
I cannot find the related problem anywhere… then I think you maybe can help.

## How to reproduce:
1. `npm i deepmerge-ts@latest`
2. write codes:
```
// lib/somefile.ts
import { deepmergeCustom } from 'deepmerge-ts'

const deepmerge = deepmergeCustom({
  mergeArrays: false,
  mergeOthers(values, utils) {
    return utils.defaultMergeFunctions.mergeOthers(values.filter(value => value !== undefined))
  },
})

export const DeepObjectMerge = <Ts extends T[], T extends Record<string, any>>(...objects: Ts) => deepmerge(...objects)

/**
 * Deep Object Plain Merge is for merging plain objects without merging methods, 
 * beware if you want to merge the class properties, use DeepObjectMerge instead
 * @param objects 
 * @returns 
 */
export const DeepObjectPlainMerge = <Ts extends T[], T extends Record<string, any>>(...objects: Ts) => {
  const plainObjects = objects.map(object => ({ ...object })) as Ts
  return deepmerge(...plainObjects)
}

// index.ts
export { DeepObjectMerge, DeepObjectPlainMerge } from 'lib/somefile.ts'
```
3. `npm run build`
4. error occurred.

## Spec:
- node: 16.16.0
- rollup: 2.78
- rollup.config.ts: same as yours...
- package.json: for main, module, types, and exports, same as yours.

Do you know what causing the error? 
Thank you

**Comment by RebeccaStevens** [comment, marked as answer]
Hey, I'll be able to have a look at this a bit later but at first glance it looks like it might be a problem with the rollup configuration. Would you be able to provide your `rollup.config` file?

**Reply by fatihaziz** [reply]
Here I reproduce the error on this repo https://github.com/fatihaziz/reproduce-rollup-err

**Reply by RebeccaStevens** [reply]
So it turns out this is an issue with the rollup typescript plugin not supporting the `.d.mts` extension. I've created a PR to address this here: https://github.com/rollup/plugins/pull/1249

**Reply by fatihaziz** [reply]
That's great! I will follow the update there.

**Reply by fatihaziz** [reply]
anyway, there is a simple fix for now until the PR accepted?

**Reply by RebeccaStevens** [reply]
If you are using yarn 3 or some other package manager that supports patching, you can either patch the rollup plugin or patch this library to use `.d.ts` instead of `.d.mts`.

But there isn't really any nice workaround right now.

**Reply by fatihaziz** [reply]
it seems the rollup plugin is updated today,
let me try again

**Reply by fatihaziz** [reply]
It working now
